### DISCUSSION TITLE
importing data from a data note

### DISCUSSION THREAD

**Discussion by jobidon** [TriliumNext/Trilium#2509] [Q&A] [answered]
I am building a script to add notes based on a json file. Once the json data is created as a new note through the GUI or by scripting using createDataNote(), how can I refer to it from my script, located in a different note?

The json file is quite large, so storing it alongside the script in the same note is not practical. Is there a way to 'import' the content of a data note by scripting?

**Comment by zadam** [comment, marked as answer]
Sure, the trivial way is to just hardcode the note ID into the script:

```javascript
const dataNote = api.getNote('bMY492A20J2L'); // hardcode noteId

const json = dataNote.getJsonContent();
```

A nicer variant is to define a relation from your script note to the data note, e.g. `~dataNote=...` and then access it like this:

```javascript
const dataNote = api.currentEntity.getRelationTarget('dataNote');

const json = dataNote.getJsonContent();
```

**Comment by jobidon** [comment]
Thanks for the reply!
The first suggestion works well, but the 2nd one doesn't:
`Cannot read property 'getRelationTarget' of undefined`

I ended up using the following:
`const dataNote = api.currentNote.children[0];`
which simply uses the first child of the note for the source data.

I am finding myself debugging a lot of backend scripts and the api.log() methd definitely has its shortcomings. Do you have a guide to describe how to setup the environment to easily step through backend code? I suspect I should install the backend separately as described [here](https://github.com/zadam/trilium/wiki/Manual-server-installation), but any other recommendation that have made your life easier as you go through the backend code?

**Reply by zadam** [reply]
Just checkout the git repo, `npm install`, `npm run start-electron` and connect the debugger on port 5858.
